### DISCUSSION TITLE
How to generate CORE_RL_*.lib files for Centos/Linux 7

### DISCUSSION THREAD

**Discussion by ptiruvaipati** [ImageMagick/ImageMagick#6939] [Help]
### ImageMagick version

7.1.1-22

### Operating system, version and so on

Centos/Linux 7

### Description

Hi,

We want to use ImageMagick 7.1.1-22 for Centos/Linux 7. 

How to generate the below files for CentOS/Linux 7 ?

CORE_RL_aom_.lib
CORE_RL_brotli_.lib
CORE_RL_bzlib_.lib
CORE_RL_cairo_.lib
CORE_RL_coders_.lib
...

We tried the below command.

./configure CC=/usr/bin/gcc CFLAGS=-08 --prefix=/usr/ImageMagick/ImageMagick-7.1.1-22

Build was successful. But it's not generated the above files.

Regards,
Pavan

**Comment by urban-warrior** [comment]
You command produces `.la` and `.so` files in `coders/.libs`.  Add `--disable-shared` to the `configure` command-line to produce `.a` files.  To produce `.lib` you need to edit the `configure.ac` script and add this statement:

```
# Output file extension
AC_SUBST([OBJEXT], ['.lib'])
```

and use `autoreconf --force --install` to regenerate the `configure` script.

Advisement from `autotoconf` says:

```
# All known linkers require a '.a' archive for static linking (except MSVC and
# ICC, which need '.lib').
libext=a
```

**Reply by ptiruvaipati** [reply]
@urban-warrior, 
Many Thanks for a quick response.

I am seeing only **.o** files in coders/.libs. 
Added "--disable-shared" to the configure command-line. 

It is still generating ".o" files. 

![image](https://github.com/ImageMagick/ImageMagick/assets/123535772/d5fea9ef-4b2d-42fa-8b1c-ff5aa46bdc0d)

Here is the command. 

./configure CC=/usr/bin/gcc --disable-shared CFLAGS=-DMAGICK_CORE_QUANTUM_DEPTH=8 PREFIX=/usr/ImageMagick/ImageMagick-7.1.1-22
